src/Agent.Listener/Configuration/CredentialProvider.cs

```diff
@@ -42,18 +45,28 @@ public CredentialProvider(string scheme)
 
     public sealed class AadDeviceCodeAccessToken : CredentialProvider
     {
-        private string _azureDevOpsClientId = "97877f11-0fc6-4aee-b1ff-febb0519dd00";
+        private VssCredentials _cacheCreds = null;
 
-        public override Boolean RequireInteractive => true;
+        //Default Application (Client) Id
+        private readonly string _clientId = "97877f11-0fc6-4aee-b1ff-febb0519dd00";
 
+        //Default ADO resource ID for user impersonation scope
+        private readonly string _userImpersonationScope = "499b84ac-1321-427f-aa17-267ca6975798/user_impersonation";
```

Comment:
Better to ask for "499b84ac-1321-427f-aa17-267ca6975798/.default" - for now that gives you the user_impersonation scope, but it's more future-proof way to get it.

Reply:
Fixed